module/VuFind/src/VuFind/Related/MoreByAuthorSolr.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+/**
+ * Related Records: Solr-based "more by author"
+ *
+ * PHP version 8
+ *
+ * Copyright (C) Villanova University 2024.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Related_Records
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:related_records_modules Wiki
+ */
+
+namespace VuFind\Related;
+
+use VuFindSearch\Command\SearchCommand;
+use VuFindSearch\Query\Query;
+
+use function count;
+
+/**
+ * Related Records: Solr-based "more by author"
+ *
+ * @category VuFind
+ * @package  Related_Records
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:related_records_modules Wiki
+ */
+class MoreByAuthorSolr implements RelatedInterface
+{
+    /**
+     * Similar records
+     *
+     * @var array
+     */
+    protected array $results = [];
+
+    /**
+     * Author being searched
+     *
+     * @var string
+     */
+    protected string $author = '';
+
+    /**
+     * Maximum number of titles to suggest
+     *
+     * @var int
+     */
+    protected int $maxRecommendations = 5;
+
+    /**
+     * Constructor
+     *
+     * @param \VuFindSearch\Service $searchService Search service
+     */
+    public function __construct(protected \VuFindSearch\Service $searchService)
+    {
+    }
+
+    /**
+     * Establishes base settings for making recommendations.
+     *
+     * @param string                            $settings Settings from config.ini
+     * @param \VuFind\RecordDriver\AbstractBase $driver   Record driver object
+     *
+     * @return void
+     */
+    public function init($settings, $driver)
+    {
+        $this->results = [];
+        if ($this->author = $driver->tryMethod('getPrimaryAuthor')) {
+            $queryStr = '"' . addcslashes($this->author, '"') . '"';
+            $query = new Query($queryStr, 'Author');
+            $command = new SearchCommand(DEFAULT_SEARCH_BACKEND, $query, 0, $this->maxRecommendations + 1);
```

Comment:
What happens if `DEFAULT_SEARCH_BACKEND` is not Solr?  

Is there anything about this recommendation module that wouldn't work with a non-Solr backend that supported `Author` as a search handler?  (Not that I know if there are any, but presumably EDS could do something similar?)  I.e. what we we didn't force this to use the default backend, and instead to use this record's backend?

Reply:
I think it's a toss-up here between `DEFAULT_SEARCH_BACKEND` and hard-coded `'Solr'`. I opted for `DEFAULT_SEARCH_BACKEND` on the theory that if we ever change that constant, we'll have to build something that's functionally equivalent to Solr, and make sure that everything works as expected with the new platform. But it might be more conservative to just say 'Solr' since 'Solr' is in the name of the class anyway.

I would not want to use the backend of the provided record, since that's guaranteed not to work for some backends, and it would break the ability to, for example, use this to display locally-indexed authors as a cross-reference from a record pulled from a third-party system.

I'm happy to change DEFAULT_SEARCH_BACKEND to 'Solr' if you think that would be better, though -- please let me know your opinion! :-)

Reply:
I think your logic is sound on leaving it as `DEFAULT_SEARCH_BACKEND`.

> I would not want to use the backend of the provided record, since that's guaranteed not to work for some backends, and it would break the ability to, for example, use this to display locally-indexed authors as a cross-reference from a record pulled from a third-party system.

I see this part a little differently.  a) I think we could support same-backend searching for those backends that support it, hiding the widget if not, and b) for cross-referencing, if we assume that the "primary" backend gets the most use, then it's more frequently helpful to display non-primary backend recommendations on primary backend record pages than the opposite.  All that said, we shouldn't do anything about this right now until there is a real use case. :)  Code is fine as is.

Reply:
Thanks! I'll merge as-is, and if other use cases crop up, I think we can pretty easily make this module configurable, so you can specify a backend via the configuration and/or toggle other behaviors.